examples/restclient/README.md

```diff
@@ -0,0 +1,79 @@
+# Quarkus RestClient
+
+Showcase demonstrating how to test a RestClient with WireMock
+
+As the reactive RestClient should be a drop in replacement for the _normal_ RestClient we use in our
+showcase only the reactive RestClient
+
+## TestResource and Annotation
+
+To separate each test responses for your testcase you need to inject the startet WireMockServer into
+your testcase.
+
+For this override the `inject` Method in your Extension and define an Annotation.
+
+```kotlin
+class WireMockExtensions : QuarkusTestResourceLifecycleManager {
+
+    private val wireMockServer = WireMockServer(
+        WireMockConfiguration.wireMockConfig().dynamicPort().httpsPort(-1).keystorePath(null)
+    )
+
+    override fun start(): MutableMap<String, String> {
+        wireMockServer.start()
+
+        return mutableMapOf("""quarkus.rest-client."cnt.adapters.restclient.MovieRestClient".url""" to wireMockServer.baseUrl())
+    }
+
+    override fun stop() {
+        wireMockServer.stop()
+    }
+
+    // Maps the Annotation InjectWireMockServer to the local variable wireMockServer
+    // and allows you in the testcase to define your responses
+    override fun inject(testInjector: TestInjector) {
+        testInjector.injectIntoFields(
+            wireMockServer,
+            AnnotatedAndMatchesType(InjectWireMockServer::class.java, WireMockServer::class.java)
+        )
+    }
+}
+```
+
+```kotlin
+@Target(FIELD, VALUE_PARAMETER)
+@Retention
+annotation class InjectWireMockServer()
+```
+
+## Attention
+
+If you specify one `@QuarkusTest` with `@QuarkusTestResource`, it will be __applied to all__ your
+Tests which have a `@QuarkusTest` annotation.
+
+This doesn't allow you to separate the TestResource for different tests.
```

Comment:
If you understand the problem correctly, test profiles could be a solution here, see https://github.com/csh0711/quarkus-testautomation/blob/master/src/test/kotlin/info/novatec/quarkus/utils/TestProfiles.kt.

Reply:
yes, they are :-)